src/metabase/api/card.clj

```diff
@@ -1144,29 +1151,31 @@ saved later when it is ready."
         filename-prefix   (or (second (re-matches #"(.*)\.csv$" filename))
                               filename)
         driver            (driver.u/database->driver database)
-        _                 (or (driver/database-supports? driver :uploads nil)
+        _check-setting    (or (driver/database-supports? driver :uploads nil)
                               (throw (Exception. (tru "Uploads are not supported on {0} databases." (str/capitalize (name driver))))))
         table-name        (->> (str (public-settings/uploads-table-prefix) filename-prefix)
-                               (upload/unique-table-name driver))
+                               (upload/unique-table-name driver)
+                               (u/lower-case-en))
         schema+table-name (if (str/blank? schema-name)
                             table-name
                             (str schema-name "." table-name))
-        _                 (upload/load-from-csv driver db-id schema+table-name csv-file)
-        _                 (sync/sync-database! database)
-        table-id          (t2/select-one-fn :id Table :db_id db-id :%lower.name table-name)]
-    (when (nil? table-id)
+        _load!            (upload/load-from-csv driver db-id schema+table-name csv-file)
+        ;; A sync is needed immediately to create the Table; the scan is settings-dependent and can be async
+        table             (sync-tables/create-or-reactivate-table! database {:name table-name :schema schema-name})]
+    (when (nil? table)
       (driver/drop-table driver db-id table-name)
       (throw (ex-info (tru "The CSV file was uploaded to {0} but the table could not be found on sync." schema+table-name)
                       {:status-code 422})))
+    (maybe-scan-table! database table)
```

Comment:
Not sure if I like this as a separate helper function. It makes it harder to read understand when the table is scanned or not, which leads to bugs like above